runtime/lua/vim/lsp/util.lua

```diff
@@ -1308,180 +1260,23 @@ function M.stylize_markdown(bufnr, contents, opts)
-  local matchers = {
-    block = { nil, '```+%s*([a-zA-Z0-9_]*)', '```+' },
-    pre = { nil, '<pre>([a-z0-9]*)', '</pre>' },
-    code = { '', '<code>', '</code>' },
-    text = { 'text', '<text>', '</text>' },
-  }
-
-  local match_begin = function(line)
-    for type, pattern in pairs(matchers) do
-      local ret = line:match(string.format('^%%s*%s%%s*$', pattern[2]))
-      if ret then
-        return {
-          type = type,
-          ft = pattern[1] or ret,
-        }
-      end
-    end
-  end
-
-  local match_end = function(line, match)
-    local pattern = matchers[match.type]
-    return line:match(string.format('^%%s*%s%%s*$', pattern[3]))
-  end
-
   -- Clean up
-  contents = M._trim(contents, opts)
-
-  local stripped = {}
-  local highlights = {}
-  -- keep track of lnums that contain markdown
-  local markdown_lines = {}
-  do
-    local i = 1
-    while i <= #contents do
-      local line = contents[i]
-      local match = match_begin(line)
-      if match then
-        local start = #stripped
-        i = i + 1
-        while i <= #contents do
-          line = contents[i]
-          if match_end(line, match) then
-            i = i + 1
-            break
-          end
-          table.insert(stripped, line)
-          i = i + 1
-        end
-        table.insert(highlights, {
-          ft = match.ft,
-          start = start + 1,
-          finish = #stripped,
-        })
-        -- add a separator, but not on the last line
-        if opts.separator and i < #contents then
-          table.insert(stripped, '---')
-          markdown_lines[#stripped] = true
-        end
-      else
-        -- strip any empty lines or separators prior to this separator in actual markdown
-        if line:match('^---+$') then
-          while
-            markdown_lines[#stripped]
-            and (stripped[#stripped]:match('^%s*$') or stripped[#stripped]:match('^---+$'))
-          do
-            markdown_lines[#stripped] = false
-            table.remove(stripped, #stripped)
-          end
-        end
-        -- add the line if its not an empty line following a separator
-        if
-          not (
-            line:match('^%s*$')
-            and markdown_lines[#stripped]
-            and stripped[#stripped]:match('^---+$')
-          )
-        then
-          table.insert(stripped, line)
-          markdown_lines[#stripped] = true
-        end
-        i = i + 1
-      end
-    end
-  end
-
-  -- Handle some common html escape sequences
-  stripped = vim.tbl_map(function(line)
-    local escapes = {
-      ['&gt;'] = '>',
-      ['&lt;'] = '<',
-      ['&quot;'] = '"',
-      ['&apos;'] = "'",
-      ['&ensp;'] = ' ',
-      ['&emsp;'] = ' ',
-      ['&amp;'] = '&',
-    }
-    return (string.gsub(line, '&[^ ;]+;', escapes))
-  end, stripped)
+  contents = M.trim_empty_lines(contents)
+  contents = vim.split(table.concat(contents, '\n'):gsub('</?pre>', '```'):gsub('\r', ''), '\n')
```

Comment:
Is the `pre` replacement still needed?

Reply:
I... think so? @clason is also suggesting we replace `<code>` tags [here](https://github.com/neovim/neovim/pull/25073#discussion_r1324855859).

Reply:
I think this was overfitting to Neovim's own docstrings, which used `<pre>` instead of traditional ```` ``` ```` code blocks. Though that will soon be rectified 😈

Reply:
If we know that servers wouldn't return that, 🔥

Reply:
If it doesn't break any tests I'd say leave it out. If servers return it I'm sure we'll get complaints very quickly and can re-add it,  including tests and comments mentioning the servers using the tags

Reply:
In general, I would recommend starting with the minimum of transformation and add them back later (only) if someone complains.

Reply:
Fair, I'll remove the `<pre>` transformation.